respect-app-compose/src/commonMain/kotlin/world/respect/app/view/report/graph/CombinedGraph.kt

```diff
@@ -0,0 +1,182 @@
+                        maxLines = 1,
+                        fontSize = MaterialTheme.typography.labelSmall.fontSize
+                    )
+                }
+            }
+        },
+        yAxisTitle = {
+            if (!isSmallSize) {
+                Box(modifier = Modifier.fillMaxHeight(), contentAlignment = Alignment.Center) {
+                    Text(
+                        text = if (reportResult.yAxisType == YAxisTypes.DURATION) {
+                            stringResource(Res.string.duration) + getDurationUnitTitle(reportResult.yRange.endInclusive)
+                        } else {
+                            stringResource(Res.string.count)
+                        },
+                        modifier = Modifier.rotateVertically(VerticalRotation.COUNTER_CLOCKWISE)
+                    )
+                }
+            }
+        }
+    ) {
+        /*
+         * Create one barchart. For each series-subgroup combination, create a series on the barchart
+         * and emit an item for each result row.
+         */
+        GroupedVerticalBarPlot {
+            reportResult.distinctSubgroups.filter {
+                it.series.reportSeriesOptions.reportSeriesVisualType == ReportSeriesVisualType.BAR_CHART
+            }.forEach { resultSubgroup ->
+                series(solidBar(colorMap[resultSubgroup] ?: Color.Transparent)) {
+                    resultSubgroup.subgroupData.forEach { resultRow ->
+                        item(resultRow.xAxis, 0f, resultRow.yAxis.toFloat())
+                    }
+                }
+            }
+        }
+
+        /*
+         * Add a line plot for each series-subgroup combination
+         *  as per https://koalaplot.github.io/0.5/docs/xygraphs/line_plots/
+         */
+        reportResult.distinctSubgroups.filter {
+            it.series.reportSeriesOptions.reportSeriesVisualType == ReportSeriesVisualType.LINE_GRAPH
+        }.forEachIndexed { subgroupIndex, resultSubgroup ->
+            LinePlot(
+                data = resultSubgroup.subgroupData.map { row ->
+                    Point(
+                        row.xAxis,
+                        row.yAxis.toFloat()
+                    )
+                },
+                lineStyle = LineStyle(
+                    brush = SolidColor(colorMap[resultSubgroup] ?: Color.Transparent),
+                    strokeWidth = 2.dp
+                ),
+                symbol = {
+                    Symbol(
+                        shape = RoundedCornerShape(4.dp),
+                        fillBrush = SolidColor(colorMap[resultSubgroup] ?: Color.Transparent)
+                    )
+                }
+            )
+        }
+    }
+}
+
+@Composable
+private fun getDurationUnitTitle(max: Float): String {
+    return when {
+        max >= MS_IN_HOUR -> " (${stringResource(Res.string.hour_unit)})"
+        else -> " (${stringResource(Res.string.minute_unit)})"
+    }
+}
+
+@Composable
+fun UiText.asString(): String {
```

Comment:
This function should not exist. We already have ```uiTextStringResource``` (in world.respect.app.components)